From: Michael Tokarev <mjt@tls.msk.ru>
Date: Sun, 10 Apr 2022 00:06:35 +0300
Subject: header: remove _PUBLIC_ define

Samba installs public headers with _PUBLIC_ markers for all externally-visible
symbols.  And it also defines _PUBLIC_ if it is not already defined.
This define should be removed, because it is in an reserved namespace, and
because it already stays in conflict with samba own headers of libraries it
is building internally.  The markers for every symbol are removed separately,
this patch removes the _PUBLIC_ definition itself.

diff --git a/talloc.h b/talloc.h
index 06e81d7..dbf35b5 100644
--- a/talloc.h
+++ b/talloc.h
@@ -36,13 +35,0 @@ extern "C" {
-/* for old gcc releases that don't have the feature test macro __has_attribute */
-#ifndef __has_attribute
-#define __has_attribute(x) 0
-#endif
-
-#ifndef _PUBLIC_
-#if __has_attribute(visibility)
-#define _PUBLIC_ __attribute__((visibility("default")))
-#else
-#define _PUBLIC_
-#endif
-#endif
-
